_posts/2024-11-26-opensearch-performance-2.17.md

```diff
@@ -0,0 +1,276 @@
+has_science_table: true
+excerpt: Learn more about the strategic enhancements and performance features that OpenSearch has delivered up to version 2.17.
+featured_blog_post: false
+featured_image: false 
+---
+
+OpenSearch has always been committed to expanding functionality, scalability, and performance. In previous performance blogs, we discussed the major improvements we've made between OpenSearch 1.0 and first [2.11](https://opensearch.org/blog/opensearch-performance-improvements/)and then [2.14](https://opensearch.org/blog/opensearch-performance-2.14/). In this blog, we'll bring you up to date on our continuing performance improvements through OpenSearch 2.17.
+
+The wide range of applications that OpenSearch supports means that no one number can summarize the improvements you will see in your applications. That's why we're reporting on a variety of performance metrics, some mostly relevant to analytics in general and log analytics in particular, some mostly relevant to lexical search, and yet others relevant to semantic search using vector embeddings and k-NN. Under the rubric of performance, we're also including improvements in resource utilization, notably RAM and disk.
+
+Overall, OpenSearch 2.17 delivers a 6x overall performance improvement over OpenSearch 1.3, with gains across essential operations such as text queries, term aggregations, range queries, date histograms, and sorting. And that's not even counting improvements to semantic vector search, which is now highly configurable to let you choose the ideal balance of response time, accuracy, and cost for your applications. All these improvements reflect the contributions and collaboration of a dedicated community, whose insights and efforts drive OpenSearch forward.
+
+This post highlights the performance improvements in OpenSearch 2.17. The first section focuses on key query operations, including text queries, term aggregations, range queries, date histograms, and sorting. These improvements were evaluated using the [OpenSearch Big5 workload](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/big5), which represents common use cases in both search and analytics applications. The benchmarks provide a repeatable framework for measuring real-world performance enhancements. The next section reports on vector search improvements. Finally, we present our roadmap for 2025, where you'll see that we're making qualitative improvements in many areas, in addition to important incremental changes. We are improving query speed by processing data in real time. We are building a query planner which uses resources more efficiently. We are speeding up intra-cluster communications. And we're adding efficient Join operations to query DSL, Piped Processing Language (PPL), and SQL. To follow our work in more detail, and to contribute comments or code, please participate on the [OpenSearch Forum](https://forum.opensearch.org/) as well as directly in our GitHub repos.
+
```

Comment:
This post covers performance improvements made through 2.17 with 1.3x as baseline. All comparisons are made against 1.3. We have given breakup of this performance against 2.14 at few places to give visibility since our last blog.

Reply:
This post covers performance improvements made through 2.17 with 1.3x as baseline. All comparisons are made against 1.3. We have given breakup of this performance against 2.14 at few places to give visibility since our last blog.